apps/wallet/src/ui/app/components/accounts/UnlockAccountContext.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { createContext, useContext, useState, type ReactNode, useCallback } from 'react';
+
+import { toast } from 'react-hot-toast';
+import { UnlockAccountModal } from './UnlockAccountModal';
+import { useAccounts } from '../../hooks/useAccounts';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { useUnlockMutation } from '../../hooks/useUnlockMutation';
+
+interface UnlockAccountContextType {
+	isUnlockModalOpen: boolean;
+	accountIdToUnlock: string | null;
+	unlockAccount: (accountId: string) => void;
+	isLoading: boolean;
+	lockAccount: (accountId: string) => void;
+	hideUnlockModal: () => void;
+}
+
+const UnlockAccountContext = createContext<UnlockAccountContextType | undefined>(undefined);
+
+export const UnlockAccountProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+	const [isUnlockModalOpen, setIsUnlockModalOpen] = useState(false);
+	const [accountIdToUnlock, setAccountIdToUnlock] = useState<string | null>(null);
+	const { data: accounts } = useAccounts();
+	const unlockAccountMutation = useUnlockMutation();
+	const backgroundClient = useBackgroundClient();
+	const hideUnlockModal = useCallback(() => {
+		setIsUnlockModalOpen(false);
+		setAccountIdToUnlock(null);
+	}, []);
+
+	const unlockAccount = useCallback(
+		async (accountId: string) => {
+			const account = accounts?.find((account) => account.id === accountId);
+			if (account) {
+				if (account.isPasswordUnlockable) {
+					// account unlocking will happen in modal after password is entered
+					setIsUnlockModalOpen(true);
+					setAccountIdToUnlock(accountId);
+				} else {
+					try {
+						// try unlocking non-password unlockable accounts immediately
+						await unlockAccountMutation.mutate({ id: accountId });
+						toast.success('Account unlocked');
```

Comment:
I found it a little strange in the screen recording where you click unlock -> the loading spinner starts -> the toast appears saying the account was unlocked -> the loading spinner disappears shortly after. It feels like the loading spinner should stop first then the toast should appear right?

I think fixing the issue from above will fix this 🤔

Reply:
fixed